# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: b79c89e6ac82
# Full Hash: b79c89e6ac82e947d6805ce7fb4ee52f28c55a5a
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
# ==============================================================================

diff -r ab5825b43bb5 -r b79c89e6ac82 js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Thu Aug 27 09:02:50 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Thu Aug 27 10:26:17 2020 +0000
@@ -233,6 +233,10 @@
 void GenerateFunctionEpilogue(jit::MacroAssembler& masm, unsigned framePushed,
                               FuncOffsets* offsets);
 
+// Iterates through frames for either possible cross-instance call or an entry
+// stub to obtain tls that corresponds to the passed fp.
+TlsData* GetNearestEffectiveTls(Frame* fp);
+
 // Describes register state and associated code at a given call frame.
 
 struct UnwindState {